backends/tfhe-cuda-backend/cuda/src/integer/compression/compression.cuh

```diff
@@ -283,88 +295,111 @@ host_integer_decompress(cudaStream_t const *streams,
     }
   }
   // Sample extract all LWEs
-  Torus lwe_accumulator_size = compression_params.small_lwe_dimension + 1;
+  auto lwe_size = compression_params.small_lwe_dimension + 1;
 
-  auto extracted_lwe = h_mem_ptr->tmp_extracted_lwe;
+  Torus *extracted_lwe;
+  if constexpr (std::is_same_v<Torus, uint64_t>) {
+    // 64 bits
+    extracted_lwe = h_mem_ptr->tmp_extracted_lwe;
+  } else {
+    // 128 bits
+    // We skip the PBS at the end
+    extracted_lwe = static_cast<Torus *>(d_lwe_array_out->ptr);
+  }
   uint32_t current_idx = 0;
   auto d_indexes_array_chunk = d_indexes_array;
   for (const auto &max_idx_and_glwe : glwe_vec) {
     const auto num_lwes = max_idx_and_glwe.first;
     extracted_glwe = max_idx_and_glwe.second;
 
-    cuda_glwe_sample_extract_64(
-        streams[0], gpu_indexes[0], extracted_lwe, extracted_glwe,
-        d_indexes_array_chunk, num_lwes, compression_params.polynomial_size,
-        compression_params.glwe_dimension, compression_params.polynomial_size);
+    if constexpr (std::is_same_v<Torus, uint64_t>)
+      cuda_glwe_sample_extract_64(streams[0], gpu_indexes[0], extracted_lwe,
+                                  extracted_glwe, d_indexes_array_chunk,
+                                  num_lwes, compression_params.polynomial_size,
+                                  compression_params.glwe_dimension,
+                                  compression_params.polynomial_size);
+    else
```

Comment:
Here we should be exhaustive I think: check if Torus is uint128_t and have a case that calls PANIC if it's neither uint64_t or uint128_t

Reply:
`host_integer_decompress` only supports 64 or 128-bit Torus. That's why we have that `static_assert` at the beginning to assert it will not be compiled for anything different than that (as `uint32_t`). So we are kind of checking it exhaustively considering the two possibilities at this point :-P

Reply:
Hmm the static_assert is only in debug and here it's a release check so maybe we should have a PANIC instead of static_assert, or we could remove it altogether and do the checks exhaustively at this point?